### PR TITLE
Generic model support

### PR DESCRIPTION
**Pull request by aryanagarwal9**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
feature

* **What is the current behavior?** (You can also link to an open issue here)
Only models specified in the model_registry can be used

* **What is the new behavior (if this is a feature change)?**
generic models, passed as a dictionary, are supported

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

* **Other information**:

### CODE REVIEW

File: src/marqo/s2_inference/s2_inference.py [hunk 1]
```diff
@@ -13,12 +15,13 @@
...
-def vectorise(model_name: str, content: Union[str, List[str]], device: str = get_default_device(), 
+
+def vectorise(model_name: str, content: Union[str, List[str]], model_properties: dict = None, device: str = get_default_device(),
                                     normalize_embeddings: bool = get_default_normalization(), **kwargs) -> List[List[float]]:
     """vectorizes the content by model name
 
     Args:
-        model_name (str): _description_
+        model_name (str) : _description_
         content (_type_): _description_
```
Comment (pandu-k) [inline src/marqo/s2_inference/s2_inference.py:26]: Please add description for model_properties

**Comment by gazambuja** [comment]
@jn2clark any ETA for this?

**Comment by jn2clark** [comment]
@gazambuja Monday 12 Dec is the scheduled next release

**Comment by pandu-k** [comment]
I ran a manual test, just using a HF model name (no model properties) There are two problems I ran into:

1. It allowed me to index all the documents, but on search an error is raised. If we can't search an index, it should prevent us indexing documents in it. A helpful error should be raised.
2. The error message does not propagate nicely to the user. It is a 500 error, without much information. I've made [this branch](https://github.com/marqo-ai/marqo/tree/catching_search_inference_errors) (feel free to merge it into this PR) which should at least help the error message propagate to the user. We need to make the error message even more helpful, such as how the user can fix the error.

Update: this error pattern also occurs when the model name is misspelled.

Code to reproduce problem:
```
mq = marqo.Client()
index_name = 'my-test-index'
model_name = 'sentence-transformers/paraphrase-albert-small-v2'

def set_up():
  try:
    mq.delete_index(index_name)
  except marqo.errors.IndexNotFoundError as e:
    pass


def generate_data(num_docs = 1000):
  vocab_source = "https://www.mit.edu/~ecprice/wordlist.10000"

  vocab = requests.get(vocab_source).text.splitlines()

  docs = [{"Title": " ".join(random.choices(population=vocab, k=10)),
           "Description": " ".join(random.choices(population=vocab, k=25)),
           "_id": f"doc_{i}"
           } for i in range(num_docs)]
  return docs


def random_1k_test():
  set_up()
  ## Set up index:
  mq.create_index(index_name=index_name, model=model_name)

  data = generate_data(num_docs=1000)
  mq.index(index_name).add_documents(data, client_batch_size=20)

  ix = mq.index(index_name)

  # ERROR OCCURS HERE:
  res = ix.search("Some string")

  top_id = res['hits'][0]['_id']
  print('results')
  pprint.pprint(res)
  print('get facet')
  pprint.pprint(ix.get_document(top_id, expose_facets=True))


random_1k_test()
```
The error that is received is an internal error. To the client it is unhelpful:
```
marqo.errors.MarqoWebError: MarqoWebError: MarqoWebError Error message: Internal Server Error
status_code: 500, type: unhandled_error_type, code: unhandled_error, link: 
```
The logs are more helpful:

```
marqo.s2_inference.errors.UnknownModelError: Could not find model properties for model=sentence-transformers/paraphrase-albert-small-v2
```

**Review by pandu-k — APPROVED** [review]

**Comment by pandu-k** [comment]
Merged mainline in again.
- Passed unit tests
- Failed DIND API tests, but all failed tests corresponded to those introduced in the visual search update

**Comment by pandu-k** [comment]
Squash merged back into mainline. Running all CI tests.
- local OS passed (besides VS tests)
- S2Search Passed (besides VS tests)
- DIND OS Passed (besides VS tests)
- Cuda DIND OS Passed (besides VS tests)
- Arm64 Local OS Passed (besides VS tests)
- Unit tests all passed
**Note:**
- VS tests == visual search API tests

**Pull request merged**